src/Symfony/Component/Translation/Translator.php

```diff
@@ -32,6 +32,8 @@ class_exists(MessageCatalogue::class);
  */
 class Translator implements TranslatorInterface, TranslatorBagInterface, LocaleAwareInterface
 {
+    use TranslatorBagTrait;
```

Comment:
Oh, Im sorry if I was unclear. 

The only method we should add to the `Translator` should be `getCatalogues`.

```php
    /**
     * @return MessageCatalogueInterface[]
     */
    public function getCatalogues(): array
    {
        return array_values($this->catalogues);
    }
```

This will make the change to the public API of `Translator` small. 

The other two methods; `all()` and `getDomains()` are only used by the providers. We could add a trait like `GetDomainsFromTranslatorTrait` (but better name) to use with `TranslationPushCommand`

```php
// Example implementation, please make sure this is working and that we are using the proper interfaces etc. 
trait GetDomainsFromTranslatorTrait 
{
    private function getDomains(Translator $translator): array
    {
        $domains = [];

        foreach ($translator->getCatalogues() as $catalogue) {
            $domains += $catalogue->getDomains();
        }

        return array_unique($domains);
    }
}
```

Same thing for the `LocoProvider` that is using `all()`. Does it make sense? And does that work?

Reply:
I've just pushed a fix for this. In fact we don't need `all()` method anymore, as we have the public `getCatalogues()`, `all()` was a shortcut, used once. I've removed it.

For `getDomains()`, the only place where it is used, is `TranslationPushCommand`, so I've moved it as a private method of the command. If needed, we could move it later in a Trait.